stdlib/LinearAlgebra/src/uniformscaling.jl

```diff
@@ -141,23 +141,23 @@ end
 # matrix breaks the hermiticity, if the UniformScaling is non-real.
 # However, to preserve type stability, we do not special-case a
 # UniformScaling{<:Complex} that happens to be real.
-function (+)(A::Hermitian{T,S}, J::UniformScaling{<:Complex}) where {T,S}
-    A_ = copytri!(copy(parent(A)), A.uplo)
-    B = convert(AbstractMatrix{Base._return_type(+, Tuple{eltype(A), typeof(J)})}, A_)
-    @inbounds for i in diagind(B)
-        B[i] += J
+function (+)(A::Hermitian, J::UniformScaling{<:Complex})
+    TS = Base._return_type(+, Tuple{eltype(A), typeof(J)})
```

Comment:
Would it be safer to do something like
```suggestion
    TS = typeof(zero(eltype(A)) + zero(eltype(J)))
```
?

Reply:
I think it's implemented in the current way because they were worried about the case where `eltype(A)` is a matrix or similar — that's why it adds `J` recursively to the diagonals — in which case `zero` doesn't work.

A safer thing that still handles this case would be
```jl
TS = isempty(A) ? Base._return_type(+, Tuple{eltype(A), typeof(J)}) : typeof(first(A) + J)
```
Or alternatively I suppose we could just throw an error for `isempty(A)`.

Reply:
I always forget about the array of arrays case...

Reply:
Also, note that the `Base._return_type` usage appears in all of the `J ± matrix` methods, so if we want to change them it seems best to do in a separate PR, especially if we want to change the behavior for empty matrices?

Reply:
I went ahead and pushed a fix to change this for all of the J ± matrix methods.